filesystem/profile.000-msys2.sh

```diff
@@ -11,8 +11,11 @@ _warn_deprecated_winver()
     if [ "$__MSYS2_WINDOWS_VERSION_WARNING_DONE" = "true" ]; then
         return;
     fi
-    local winver;
-    winver="$(uname -s | sed -ne 's/\([^-]*\)-\([^-]*\).*/\2/p')"
+
+    local uname_s
+    uname_s="$(uname -s)"
+    local parts=(${uname_s//-/ })
+    local winver="${parts[1]}"
```

Comment:
Wouldn't it be slightly easier to read as:

```sh
local winver
winver=$(uname -s)   # looks like `MINGW64_NT-10.0-22621`
winver=${winver#*-}  # strip off `<prefix>-`
winver=${winver%%-*} # strip off `-<suffix>`, if any
```

As a bonus, this would be pure POSIX shell (apart from the `local` construct).